File: src/python/core/src/dicebear/style_def/meta_creator.py
"""Read-only view over the ``meta.creator`` block of a style definition."""

from __future__ import annotations

from typing import Any, cast


class MetaCreator:
    """Read-only view over the ``meta.creator`` block of a style definition."""

    def __init__(self, data: dict[str, Any]) -> None:
        self._data = data

    def name(self) -> str | None:
        """Return the creator's display name, or ``None`` when not set."""
        return cast("str | None", self._data.get("name"))

    def url(self) -> str | None:
        """Return the creator's homepage URL, or ``None`` when not set."""
        return cast("str | None", self._data.get("url"))

File: src/csharp/core/src/Style/MetaCreator.cs
using System.Text.Json.Nodes;

namespace DiceBear.Internal
{
    /// <summary>
    /// Read-only view over the <c>meta.creator</c> block of a style
    /// definition.
    /// </summary>
    internal sealed class MetaCreator
    {
        private readonly JsonObject _data;

        internal MetaCreator(JsonObject data) => _data = data;

        /// <summary>
        /// Returns the creator's display name, or <see langword="null"/> when
        /// not set.
        /// </summary>
        internal string? Name() => JsonRead.Str(_data, "name");

        /// <summary>
        /// Returns the creator's homepage URL, or <see langword="null"/> when
        /// not set.
        /// </summary>
        internal string? Url() => JsonRead.Str(_data, "url");
    }
}
